sarcophagus-org/sarcophagus-v2-contracts
# Add hardhat task to seed contract with data

# Add hardhat task to seed data to contract 

Adds a hardhat task called `generate-history` that seeds the contract with archaeologists, accused sarcophagi, and successful sarcophagi. 

```
npx hardhat generate-history --network localhost
```

### Config
Use the config file in `tasks/generate-history/config.ts` to define config values. 
- archaeologistCount: The number of archaeologists to register. This has nothing to do with the archaeologist service, it only calls register on the contract.
- sarcohpagusCount: The number of sarcophagi to create using random archaeologist that are registered and random values for minshards/requiredArchs.
- accusedSarcophagusCount: the number of sarcophagi to call accuse on. Calling this includes all archaeologists on the sarcophagus. 
- archaeologistUnwrapChance: The chance that an archaeologist will unwrap their sarcophagi. Depending on the config  values archs will likely have multiple sarcos. If the roll of the dice dictate that an arch will not unwrap, then they will not unwrap any of their sarcophagi. 

### Testing
1. Spin up a node 
```
npx hardhat node
```
2. Run the script. Don't forget to include `--network localhost`.
```
npx hardhat generate-history --network localhost
```
This will register archs, create sarcos, accuse archs on sarcos, and then unwrap sarcos.

3. Test by opening the console in another terminal
```
npx hardhat console --network localhost
```
4. Run the following functions in the console to see that arch profiles were created
```
const diamond = await ethers.getContract("Diamond_DiamondProxy")
const view = await ethers.getContractAt("ViewStateFacet", diamond.address)
const addresses = await view.getArchaeologistProfileAddresses()
const profiles = await view.getArchaeologistProfiles(addresses)
const stats = await view.getArchaeologistProfiles(addresses)
```
